tutorials/preprocessing/plot_60_maxwell_filtering_sss.py

```diff
@@ -106,18 +110,96 @@
 raw.info['bads'] = []
 raw_check = raw.copy()
 raw_check.pick_types(meg=True, exclude=()).load_data().filter(None, 40)
-auto_noisy_chs, auto_flat_chs = mne.preprocessing.find_bad_channels_maxwell(
+auto_noisy_chs, auto_flat_chs, auto_scores = find_bad_channels_maxwell(
     raw_check, cross_talk=crosstalk_file, calibration=fine_cal_file,
-    verbose=True)
+    return_scores=True, verbose=True)
 print(auto_noisy_chs)  # we should find them!
 print(auto_flat_chs)  # none for this dataset
-raw.info['bads'].extend(auto_noisy_chs + auto_flat_chs)
 
 ###############################################################################
-# But this algorithm is not perfect. For example, it misses ``MEG 2313``,
-# which has some flux jumps, because there are not enough flux jumps in the
-# recording. So it can still be useful to manually inspect and mark bad
-# channels:
+# Now we can update the list of bad channels in the dataset. We first create a
+# set (a collection of unique values) to ensure no channel appears more than
+# once in the list.
+
+bads = set([*raw.info['bads'], *auto_noisy_chs, *auto_flat_chs])
+bads = list(bads)
+raw.info['bads'] = bads
+
+###############################################################################
+# We called `~mne.preprocessing.find_bad_channels_maxwell` with the optional
+# keyword argument ``return_scores=True``, causing the function to return a
+# dictionary of all data related to the scoring used to classify channels as
+# noisy or flat. This information can be used to produce diagnostic figures.
+#
+# In the following, we will generate a total of four of such visualizations for
+# the automated detection of *noisy* channels.
+
+for ch_type in ('mag', 'grad'):
+    # Only select the data for mag or grad channels.
+    ch_subset = auto_scores['ch_types'] == ch_type
+    ch_names = auto_scores['ch_names'][ch_subset]
+    scores = auto_scores['scores_noisy'][ch_subset]
+    limits = auto_scores['limits_noisy'][ch_subset]
+    # Number of time windows
+    segments = range(1, len(auto_scores['bin_edges']))
```

Comment:
Can you instead use something like (after refactoring according to the above):
```
t_centers = (start + stop) / 2.
```
Then the plot would go according to time rather than segments? Not sure if this would actually work, but if there are gaps in the data (e.g., if there are BAD annotations) it will actually match the time course better.

Personally I would probably work through how to use `plt.pcolormesh` with bin edges to represent the chunks correctly, but if there are gaps in the data will require filling in gaps with `np.nan` or so, so might not be worth the trouble.